pkg/cmd/server/start/start_master.go

```diff
@@ -242,10 +242,10 @@ func (o MasterOptions) RunMaster() error {
 			return err
 		}
 
-		if err := os.MkdirAll(path.Dir(o.MasterArgs.GetConfigFileToWrite()), os.FileMode(0755)); err != nil {
```

Comment:
They should and we can control that for files we create either via openshift or via the installer. But I don't know of a way to control that for files created by hand so it seemed reasonable to set the directory up to protect against that.